[DOCS] Replace deprecated --template option with --index-management (#12882)
elastic/beats: docs/devguide/newdashboards.asciidoc

./metricbeat setup
-------------------------

The `setup` phase loads several dependencies, such as: 

- Index mapping template in Elasticsearch
- Kibana dashboards
- Machine Learning (ML) jobs
- Ingest pipelines
- ILM policy

> 

Author: ```suggestion

The actual list varies depending on the Beat you're setting up.
```

./metricbeat setup
-------------------------

The `setup` phase loads several dependencies, such as: 

- Index mapping template in Elasticsearch
- Kibana dashboards
- Machine Learning (ML) jobs
- Ingest pipelines
- ILM policy

The dependencies vary depending on the Beat you're setting up.

> For more details about the `setup` command, run:

Author: ```suggestion
For more details about the `setup` command, see the command-line help. For example:
```

> The `setup` phase loads:

Author: ```suggestion
The `setup` phase loads several dependencies, such as: 
```

- Index mapping template in Elasticsearch

> - Kibana dashboards (where available)

Author: ```suggestion
- Kibana dashboards
```

> - Machine Learning (ML) jobs (where available)

Author: ```suggestion
- Machine Learning (ML) jobs
```

> - Ingest pipelines (where available)

Author: ```suggestion
- Ingest pipelines
```

> - ILM policy (for Elasticsearch 6.5 and newer)

Reviewer: Actually, not sure about the ES version here, afaik it should be >= 6.7 (supported) or >= 7.0 (default).

Author: Actually I'm not sure it's necessary to mention the version number at all here.

Author: @simitt ^^ The version info shown here is taken directly from the output of `help setup`. I'm not sure it's necessary here, though, so Ill will change the lead-in wording in this section.

> - ILM policy (for Elasticsearch 6.5 and newer)

Author: ```suggestion
- ILM policy
```

 * Index mapping template in Elasticsearch to ensure fields are mapped.
 * Kibana dashboards (where available).
 * ML jobs (where available).
 * Ingest pipelines (where available).

>  * ILM policy (for Elasticsearch 6.5 and newer).

Author: @simitt Note that this the exact output of running `metricbeat help setup` (mentions 6.5), so I don't think I should change the version here.

Reviewer: Thanks @dedemorton, afaik it was _possible_ to import the policies from 6.5 on (beta), but one would need to set `setup.ilm.enabled: true` as with `auto` it only enables and therefore loads policies for ES >= `7.0`. I think it is ok to stick with this text, as it doesn't specify any further.

  -h, --help               help for setup
      --machine-learning   Setup machine learning job configurations only
      --modules string     List of enabled modules (comma separated)

>       --index-management           Setup index template only

Reviewer: Please also update the explanation to something like `Setup index management only (templates and ILM policies)`.

Author: Looking at this more closely, if we're going to show the output of running  `metricbeat help setup`, the example should show the exact output of running this command.
